File: src/main/java/Model/Utente/UtenteValidator.java
package Model.Utente;

import Model.Gestione.RequestValidator;

import javax.servlet.http.HttpServletRequest;
import java.util.regex.Pattern;

final class UtenteValidator {
    static RequestValidator validateForm(HttpServletRequest request,boolean update) {
        RequestValidator validator = new RequestValidator(request);
        validator.assertMatch("name", Pattern.compile("^\\w{4,20}$"),"il nome utente deve iniziare con una lettera e  avere lunghezza compresa tra i 4 e i 20 caratteri");
        if(update){
            validator.assertInt("id","Id deve essere un numero intero");
        }
        return validator;
    }

    public static RequestValidator validateSignIn(HttpServletRequest request) {
        RequestValidator validator = new RequestValidator(request);
        validator.assertEmail("email","Inserire l'email corrispondente al tuo account");
        validator.assertPassword("password", "Inserire la password corrispondente al tuo account");
        System.out.println("email " + request.getParameter("email"));
        System.out.println("password " + request.getParameter("password"));
        return validator;
    }
}

File: src/main/webapp/javascript/utenteValidator.js
let errors=[];

function validateForm(request, update)
{
    let validator = new RequestValidator(request);
    let bool;
    let boolbis;
    bool=assertMatch("name", new RegExp("^\\w{4,20}$"),"il nome utente deve iniziare con una lettera e  avere lunghezza compresa tra i 4 e i 20 caratteri");
    if(update)
    {
        boolbis=assertInt("id","Id deve essere un numero intero");
    }
    return validator;
}

function validateSignIn(request)
{
    let validator = new RequestValidator(request);
    validator.assertEmail("email","Inserire l'email corrispondente al tuo account");
    return validator;
}

function RequestValidator(request)
{
    let requestx = request;
    return requestx;
}

function gatherError(condition, message)
{
    if(condition)
    {
        return true;
    }
    else
    {
        errors.push(message);
        return false;
    }
}

function isBlank(value)
{
    //verifica se è vuota o ha solo spazi bianchi
    return (value.length === 0 || !value.trim());
}

function required(value)
{
    return  value != null && isBlank(value);
}

function assertMatch(value, regexp, msg)
{
    let param = value;
    condition.matcher=param;
    let condition = required(param) && regexp.matcher(param).matches();
    return  gatherError(condition,msg);
}

function assertInt(value, msg)
{
    let INT_PATTERN= new RegExp("^\\d+$");
    return assertMatch(value,INT_PATTERN,msg);
}

function assertEmail(value, msg)
{
    let pattern = new RegExp("^[a-zA-Z0-9.!#$%&´*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$");
    return assertMatch(value,pattern,msg);
}